Zend/zend_execute.c

```diff
@@ -1546,84 +1545,14 @@ ZEND_API ZEND_COLD void zend_wrong_string_offset_error(void)
 		case ZEND_ASSIGN_STATIC_PROP_OP:
 			msg = "Cannot use assign-op operators with string offsets";
 			break;
+		case ZEND_FETCH_DIM_UNSET:
+			msg = "Cannot unset string offsets";
+			break;
 		case ZEND_FETCH_DIM_W:
 		case ZEND_FETCH_DIM_RW:
 		case ZEND_FETCH_DIM_FUNC_ARG:
-		case ZEND_FETCH_DIM_UNSET:
 		case ZEND_FETCH_LIST_W:
-			/* TODO: Encode the "reason" into opline->extended_value??? */
```

Comment:
I also think that the old messages are better. Can we use the approach proposed above?
e.g. compiler may just store opcode of the next opline in FETCH_DIM_W->execute_data.